force-app/main/default/classes/OpportunityTriggerHandler.cls

```diff
@@ -0,0 +1,55 @@
+public with sharing class OpportunityTriggerHandler {
+
+    public void beforeInsert(List<Opportunity> newOpps) {
+
+    }
+
+    public void beforeUpdate(List<Opportunity> newOpps, Map<Id, Opportunity>  oldOppMap) {
+        //Validate Amount < 5000
+        for (Opportunity opp : newOpps) {
+            if (opp.Amount < 5000) {
+                opp.addError('Opportunity amount must be greater than 5000');
+            }
```

Comment:
Whenever I am working in the context of an update Trigger. I will check to see if the value I care about has changed. Typically, in a business, I would assume that we also had the rule to prevent an opportunity from being created with an amount of less than 5000.  So I will only check if that field changed. My answer is below. 

```
  List<Opportunity> changedOpportunities = getChangedRecords(opportunities, Trigger.oldMap, Opportunity.Amount);
        if(!changedOpportunities.isEmpty()){
           for (Opportunity opp : changedOpportunities){
               if(opp.Amount <= 5000){
                   opp.addError('Opportunity amount must be greater than 5000');
               }
           }
        }
        ```